controllers/admin/AdminCarriersController.php

```diff
@@ -642,13 +642,14 @@ public function changeZones($id)
      */
     public function getList($id_lang, $order_by = null, $order_way = null, $start = 0, $limit = null, $id_lang_shop = false)
     {
-        parent::getList($id_lang, $order_by, $order_way, $start, $limit, $id_lang_shop);
+        // Replace "0" by the name of the shop directly in SQL query (allowing sort without errors)
+        $this->_select = empty($this->_select) ? '' : ', ';
```

Comment:
You're replacing `$this->_select` with `', '`, this is way is clearer.
```suggestion
        if (!empty($this->_select)) {
            $this->_select .= ', ';
        }
```